Core/src/org/sleuthkit/autopsy/communications/snapshot/CommSnapShotReportWriter.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Set;
+import javax.imageio.ImageIO;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.report.uisnapshot.UiSnapShotReportWriter;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.CommunicationsFilter;
+import org.sleuthkit.datamodel.CommunicationsFilter.AccountTypeFilter;
+import org.sleuthkit.datamodel.CommunicationsFilter.DateRangeFilter;
+import org.sleuthkit.datamodel.CommunicationsFilter.SubFilter;
+
+/**
+ * Generate and write the Communication snapshot report to disk.
+ */
+public class CommSnapShotReportWriter extends UiSnapShotReportWriter {
+
+    private final BufferedImage image;
+    private final CommunicationsFilter filter;
+
+    /**
+     * Constructor
+     *
+     * @param currentCase      The Case to write a report for.
+     * @param reportFolderPath The Path to the folder that will contain the
+     *                         report.
+     * @param reportName       The name of the report.
+     * @param generationDate   The generation Date of the report.
+     * @param snapshot         A snapshot of the view to include in the report.
+     */
+    public CommSnapShotReportWriter(Case currentCase, Path reportFolderPath, String reportName, Date generationDate, BufferedImage snapshot, CommunicationsFilter filter) {
+
+        super(currentCase, reportFolderPath, reportName, generationDate);
+
+        this.image = snapshot;
+        this.filter = filter;
+
+    }
+
+    /**
+     * Generate and write the html page that shows the snapshot and the state of
+     * the CommunicationFilters
+     *
+     * @throws IOException If there is a problem writing the html file to disk.
+     */
+    @Override
+    protected void writeSnapShotHTMLFile() throws IOException {
+        SimpleDateFormat formatter = new SimpleDateFormat("MMMMM dd, yyyy");
+
+        ImageIO.write(image, "png", getReportFolderPath().resolve("snapshot.png").toFile()); //NON-NLS
+
+        //make a map of context objects to resolve template paramaters against
+        HashMap<String, Object> snapShotContext = new HashMap<>();
+        snapShotContext.put("reportTitle", getReportName()); //NON-NLS
+
+        List<SubFilter> filters = filter.getAndFilters();
+
+        for (SubFilter f : filters) {
+            if (f.getClass().getName().equals(DateRangeFilter.class.getName())) {
+                long startDate = ((DateRangeFilter) f).getStartDate();
+                long endDate = ((DateRangeFilter) f).getEndDate();
+
+                if (startDate > 0) {
+
+                    snapShotContext.put("startTime", formatter.format(new Date((Instant.ofEpochSecond(startDate)).toEpochMilli()))); //NON-NLS
+                }
+
+                if (endDate > 0) {
+                    snapShotContext.put("endTime", formatter.format(new Date((Instant.ofEpochSecond(endDate)).toEpochMilli()))); //NON-NLS
+                }
+            } else if (f.getClass().getName().equals(AccountTypeFilter.class.getName())) {
```

Comment:
again, I think using instanceOf is the usual idiom for this.

Reply:
This is still weird to me.  My quick research indicated that instanceOf is quite fast, and that comparing by class name is potentially dangerous (although probably not in our use case).  I wouldn't worry about performance here anyways cause it is not a huge list or we are checking or something that is done very often.  @rcordovano do you have an opinion